File: android/src/main/java/com/azifacemobile/theme/IdFeedback.java
package com.azifacemobile.theme;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

public class IdFeedback extends ViewStyle {
  private static final String KEY = "idFeedback";
  final Theme theme;
  final JSONObject target;
  final Color color;
  final Image image;

  public IdFeedback(ReactApplicationContext context, JSONObject target) {
    super(target, KEY);

    this.theme = new Theme();

    this.target = this.theme.getTarget(target, KEY);
    this.color = new Color();
    this.image = new Image(context, this.target);
  }

  @Override
  public int getForegroundColor() {
    return super.getForegroundColor("#272937");
  }

  public boolean getDisableIDFeedbackScreen() {
    final String key = "isDisableIDFeedbackScreen";
    final boolean defaultDisableIDFeedback = false;
    try {
      if (!this.theme.exists(this.target, key)) {
        return defaultDisableIDFeedback;
      }

      return this.target.getBoolean(key);
    } catch (JSONException e) {
      return defaultDisableIDFeedback;
    }
  }

  public double getDisplayTime() {
    final double defaultDisplayTime = 2.0d;
    final String key = "displayTime";
    try {
      if (!this.theme.exists(this.target, key)) {
        return defaultDisplayTime;
      }

      return this.target.getDouble(key);
    } catch (JSONException e) {
      return defaultDisplayTime;
    }
  }

  public int getFlipIDBackImage() {
    return this.image.getSource("flipIDBackImage");
  }

  public int getFlipIDFrontImage() {
    return this.image.getSource("flipIDFrontImage");
  }
}

File: ios/Theme/IdFeedback.swift
public class IdFeedback: ViewStyle {
  private static let KEY: String = "idFeedback"
  private let style: Style
  private let target: NSDictionary?
  private let color: Color
  private let image: Image

  init(target: NSDictionary?) {
    self.style = Style()

    self.target = self.style.getTarget(target, key: IdFeedback.KEY)
    self.color = Color()
    self.image = Image(target: self.target)

    super.init(target: self.target, key: IdFeedback.KEY)
  }

  override func getForegroundColor() -> UIColor {
    return super.getForegroundColor(defaultColor: "#272937")
  }

  public func getDisableIDFeedbackScreen() -> Bool {
    let key = "isDisableIDFeedbackScreen"
    if !self.style.exists(self.target, key: key) {
      return false
    }

    return self.target?[key] as! Bool
  }

  public func getDisplayTime() -> Double {
    let defaultDisplayTime: Double = 2.0
    let key = "displayTime"

    if !self.style.exists(self.target, key: key) {
      return defaultDisplayTime
    }

    let displayTime = self.target?[key] as! Double
    return displayTime < 0 ? defaultDisplayTime : displayTime
  }

  public func getFlipIDBackImage() -> UIImage? {
    return self.image.getSource("flipIDBackImage")
  }

  public func getFlipIDFrontImage() -> UIImage? {
    return self.image.getSource("flipIDFrontImage")
  }
}
